chainer/training/extensions/parameter_statistics.py

```diff
@@ -0,0 +1,146 @@
+                                                     50, 84.13, 97.72,
+                                                     99.87))
+    }
+
+    def __init__(self, links, statistics=default_statistics,
+                 report_params=True, report_grads=True, prefix=None,
+                 trigger=(1, 'epoch')):
+
+        if not isinstance(links, (list, tuple)):
+            links = links,
+        self._links = links
+
+        if statistics is None:
+            statistics = {}
+        self._statistics = statistics
+
+        attrs = []
+        if report_params:
+            attrs.append('data')
+        if report_grads:
+            attrs.append('grad')
+        self._attrs = attrs
+
+        self._prefix = prefix
+        self._trigger = training.trigger.get_trigger(trigger)
+        self._summary = reporter.DictSummary()
+
+    def __call__(self, trainer):
+        """Execute the statistics extension.
+
+        Collect statistics for the current state of parameters.
+
+        Note that this method will merely update its statistic summary, unless
+        the internal trigger is fired. If the trigger is fired, the summary
+        will also be reported and then reset for the next accumulation.
+
+        Args:
+            trainer (~chainer.training.Trainer): Associated trainer that
+                invoked this extension.
+        """
+        statistics = {}
+
+        for link in self._links:
+            link_name = getattr(link, 'name', 'None')
+            for param_name, param in link.namedparams():
+                for attr_name in self._attrs:
+                    for function_name, function in \
+                            six.iteritems(self._statistics):
+                        # Get parameters as a flattend one-dimensional array
+                        # since the statistics function should make no
+                        # assumption about the axes
+                        params = getattr(param, attr_name).ravel()
+                        value = function(params)
+                        key = self.report_key_template.format(
+                            link_name=link_name,
+                            param_name=param_name,
+                            attr_name=attr_name,
+                            function_name=function_name
+                        )
+                        if hasattr(value, '__iter__'):
+                            # Append integer indices to the keys if the
+                            # statistic function return multiple values
+                            statistics.update({'{}/{}'.format(key, i): v for
+                                               i, v in enumerate(value)})
+                        else:
+                            statistics[key] = value
+
+        # Post-process
+        if self._prefix is not None:
+            statistics = {'{}/{}'.format(self._prefix, k): v for k, v in
+                          six.iteritems(statistics)}
```

Comment:
I think we can add prefix when creating `key` in l.108 and need not to do it as a post-processing.